src/content_script.js

```diff
@@ -173,120 +169,176 @@ function createBadgeFragment (socialAction) {
-    htmlBadgeFragmentPromptButtonDiv.className = "fbc-badge-prompt-buttons";
+    // Flip the iframe to show on the left side when icon is too close to the edge
+    if (iframePaddingAllowance > calculateOffsetDiff) {
+      iframeBox.style.marginLeft = `${xLeft}px`;
+      iframeChevron.classList.add("fbc-chevron-arrow-right");
+      iframeChevron.style.marginLeft = `${xPosChevron - fencePos.offsetWidth - iframeChevron.offsetWidth - offsetX}px`;
+    }
 
-    for (let buttonString of htmlEmailBadgeFragmentPromptButtonStrings) {
-      const button = document.createElement("button");
-      button.className = "fbc-badge-prompt-" + buttonString;
-      button.appendChild(document.createTextNode( browser.i18n.getMessage(buttonString) ));
-      htmlBadgeFragmentPromptButtonDiv.appendChild(button);
+    else {
+      iframeChevron.classList.remove("fbc-chevron-arrow-right");
     }
 
-    htmlBadgeFragmentPromptDiv.appendChild(htmlBadgeFragmentPromptButtonDiv);
+    iframeChevron.classList.remove("fbc-chevron-arrow-top");
   }
 
-  // Create Empty Wrapper Div
-  const htmlBadgeWrapperDiv = document.createElement("div");
-  htmlBadgeWrapperDiv.appendChild(htmlBadgeFragment);
+  // Mobile Orientation
+  else {
+    for (const panels of iframeElement) {
+      panels.width = window.innerWidth;
+      if (window.innerWidth > 480) {
+        panels.width = 350;
+      }
+    }
 
-  return htmlBadgeWrapperDiv;
-}
+    iframeChevron.classList.add("fbc-chevron-arrow-top");
+    iframeBox.style.marginTop = `${yPosMobile}px`;
+
+    const xPosChevronMobile = xPosMobile;
+    const yPosChevronMobile = yPosMobile - iframeChevron.offsetWidth;
+
+    iframeChevron.style.marginLeft = `${xPosChevronMobile}px`;
+    iframeChevron.style.marginTop = `${yPosChevronMobile}px`;
 
-function shouldBadgeBeSmall(ratioCheck, itemHeight) {
-  if (ratioCheck < 1.1) {
-    return true;
-  } else if (itemHeight < 39) {
-    return true;
   }
-  return false;
 }
 
-function openLoginPrompt(parent, el, htmlBadgeDiv) {
-  parent.classList.toggle("active");
-  positionPrompt( htmlBadgeDiv );
-  el.classList.toggle("js-fbc-prompt-active");
-  document.body.classList.toggle("js-fbc-prompt-active");
-  htmlBadgeDiv.querySelector(".fbc-badge-prompt-btn-cancel").focus();
+function openLoginPrompt(socialAction, fencePos, target, FBC_IFRAME_HEIGHT) {
+  const hasFbcWrapper = document.querySelector(".fbc-wrapper");
+  if (!hasFbcWrapper) {
+    injectIframeOntoPage(socialAction, target, FBC_IFRAME_HEIGHT);
+    positionIframe(fencePos);
+
+    ["resize", "scroll"].forEach(function (evt) {
+      if (document.querySelector(".fbc-wrapper")) {
+        window.addEventListener(evt, () => {
+          positionIframe(fencePos);
+        });
+      }
+    });
+
+    window.addEventListener("message", (e) => {
+      if (
+        e.data === "allowTriggered" 
+        && e.origin === "moz-extension://ad96861a-bb7a-4c63-a924-2f21045b80aa"
+      ){
+        target.click();
+      }
+    });
```

Comment:
@groovecoder I did more research on the `Window.postMessage` method and it states that if I specify the origin where the data is coming from (in this case, the `iframe`) it should be fine.

From [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage)
> If you do expect to receive messages from other sites, always verify the sender's identity using the origin and possibly source properties.